cmd/repository.go

```diff
@@ -0,0 +1,686 @@
+	}
+	return cmd
+}
+
+func NewRepositoryRenameCmd(clientFn repositoryClientFn) *cobra.Command {
+	cmd := &cobra.Command{
+		Short:   "Rename a repository",
+		Use:     "rename <old> <new>",
+		PreRunE: bindEnv("repositories", "confirm"),
+	}
+
+	cmd.Flags().BoolP("confirm", "c", false, "Prompt to confirm all options interactively (Env: $FUNC_CONFIRM)")
+	cmd.Flags().StringP("repositories", "r", repositoriesPath(), "Path to language pack repositories (Env: $FUNC_REPOSITORIES)")
+
+	cmd.RunE = func(_ *cobra.Command, args []string) error {
+		return runRepositoryRename(args, clientFn)
+	}
+	return cmd
+}
+
+func NewRepositoryRemoveCmd(clientFn repositoryClientFn) *cobra.Command {
+	cmd := &cobra.Command{
+		Short:      "Remove a repository",
+		Use:        "remove <name>",
+		Aliases:    []string{"rm"},
+		SuggestFor: []string{"delete", "del"},
+		PreRunE:    bindEnv("repositories", "confirm"),
+	}
+
+	cmd.Flags().BoolP("confirm", "c", false, "Prompt to confirm all options interactively (Env: $FUNC_CONFIRM)")
+	cmd.Flags().StringP("repositories", "r", repositoriesPath(), "Path to language pack repositories (Env: $FUNC_REPOSITORIES)")
+
+	cmd.RunE = func(_ *cobra.Command, args []string) error {
+		return runRepositoryRemove(args, clientFn)
+	}
+	return cmd
+}
+
+// client config
+// -------------
+
+// repositoryClientConfig used for instantiating a fn.Client
+type repositoryClientConfig struct {
+	Repositories string // Path to repos to be managed
+	Verbose      bool   // Enables verbose logging
+	Confirm      bool   // Enables interactive confirmation/prompting mode
+}
+
+// newRepositoryConfig creates a configuraton suitable for use instantiating the
+// fn Client. Note that parameters for the individual commands (add, remove etc)
+// are collected separately in their requisite run functions.
+func newRepositoryConfig(args []string) (cfg repositoryClientConfig, err error) {
+	// inital config is populated based on flags, which are themselves
+	// first populated by static defaults, then environment variables,
+	// finally command flags.
+	cfg = repositoryClientConfig{
+		Repositories: viper.GetString("repositories"),
+		Verbose:      viper.GetBool("verbose"),
+		Confirm:      viper.GetBool("confirm"),
+	}
+
+	// If not in confirm (interactive prompting) mode,
+	// this struct is complete.
+	if !cfg.Confirm {
+		return
+	}
+
+	// Prompt the terminal for interactive input using the current values
+	// as defaults. (noninteractive terminals are a noop)
+	if interactiveTerminal() {
+		return cfg.prompt()
+	}
+
+	// Noninteractive terminals in confirm/prompt mode simply echo
+	// effective values to stdout.
+	fmt.Fprintf(os.Stdout, "Repositories path: %v\n", cfg.Repositories)
+	fmt.Fprintf(os.Stdout, "Verbose logging:   %v\n", cfg.Verbose)
+	return
+}
+
+// prompt returns a config with values populated from interactivly prompting
+// the user.
+func (c repositoryClientConfig) prompt() (repositoryClientConfig, error) {
+	if !interactiveTerminal() {
```

Comment:
I don't think this will ever be true, since `prompt()` is only ever called from `newRepositoryConfig()` which checks for this first.

Reply:
👍🏻 I waffled about whether or not to leave this in there despite the redundancy, eventually opting to leave it for excessive clarity to the reader; but if you also notice this is strictly unnecessary, then that breaks my mental tie.  The check is now removed from `prompt`, presuming the caller will already have checked (even on future refactors)